src/core/positioning/udpreceiver.cpp

```diff
@@ -35,8 +35,10 @@ UdpReceiver::UdpReceiver( const QString &address, const int port, QObject *paren
   mSocket->setSocketDescriptor( sockfd, QUdpSocket::UnconnectedState );
 #endif
 
-  connect( mSocket, &QAbstractSocket::stateChanged, this, &UdpReceiver::setSocketState );
   connect( mSocket, qOverload<QAbstractSocket::SocketError>( &QAbstractSocket::errorOccurred ), this, &UdpReceiver::handleError );
+  connect( mSocket, &QUdpSocket::stateChanged, this, [=]( QAbstractSocket::SocketState state ) {
```

Comment:
Again here I think you can just connect to the abstract gnss receiver class' function.